repos/spack_repo/builtin/packages/py_rasterio/package.py

```diff
@@ -75,13 +82,14 @@ class PyRasterio(PythonPackage):
         depends_on("py-setuptools", when="@:1.3.9")
         depends_on("py-snuggs@1.4.1:", when="@:1.3")
 
-    # From README.rst and setup.py
-    depends_on("gdal@3.5:", when="@1.4:")
-    depends_on("gdal@3.1:", when="@1.3:")
-    depends_on("gdal@2.4:3.3", when="@1.2.7:1.2")
-    depends_on("gdal@2.3:3.2", when="@1.2.0:1.2.6")
-
     # https://github.com/rasterio/rasterio/issues/3371
     conflicts("^gdal@3.11:", when="@:1.4.3")
     # https://github.com/rasterio/rasterio/pull/3212
     conflicts("^gdal@3.10:", when="@:1.4.1")
+
+    # ensure cython absolutely gets the right gdal and embeds the correct rpaths
+    def setup_build_environment(self, env):
+        gdal = self.spec["gdal"]
+        # looks for this envar in setup.py
+        env.set("GDAL_CONFIG", join_path(gdal.prefix.bin, "gdal-config"))
+        env.prepend_path("LDFLAGS", f"-Wl,-rpath,{gdal.libs.directories[0]}")
```

Comment:
We used to use `self.compiler.cc_rpath_arg` for this, but NAG was the only compiler that was different and we removed support for NAG.

Reply:
I don't understand this comment. Is it a FYI for me, or something you want changed?